### PR TITLE
Add workflow to spell check PRs

### PR DESCRIPTION
**Pull request by svrnm**
Do we want to introduce a workflow like this one that spell checks PRs to the community repo? 

For me as a non-native English speaker having spellchecking for repositories I regularly commit to, is a huge help, especially if I use the github editor for some quick changes, or if I review PRs where I don't see them easily. 

[cspell](https://cspell.org/) is one of many options to do so, we use it for the opentelemetry.io repository successfully: the workflow suggested here is able to annotate files within PRs making it super easy to find necessary changes.

Note: This PR still needs tweaking since there are plenty of unknown words (especially names or github handles), but I would take the time & effort to add them all to the `.cspell.yaml` as needed

### CODE REVIEW

**Review by arminru — APPROVED** [review]
LGTM, thank you!

**Comment by svrnm** [comment]
> Note: This PR still needs tweaking since there are plenty of unknown words (especially names or github handles), but I would take the time & effort to add them all to the .cspell.yaml as needed

The .cspell.yaml file is now up-to-date, all that is missing are typos (see #1811)

File: .cspell.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    - tigran
+    - tigrannajaryan
+    - trask
+    - tsloughter
+    - tylerbenson
+    - xoscar
+    - yahn
+    # slack spaces
+    - otel-agentmanwg
+    - otel-ebpf
```
Comment (danielgblanco) [inline .cspell.yaml:122]: Should newlines be added here and below?

**Review by danielgblanco — APPROVED** [review]
This can be very useful, thanks! I left a small comment but looks good

**Review by trask — APPROVED** [review]

**Pull request merged**